internal/metrics/collector.go

```diff
@@ -0,0 +1,267 @@
+package metrics
+
+import (
+	"context"
+
+	"github.com/octopusdeploy/octopus-permissions-controller/api/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type MetricsCollector struct {
+	client client.Client
+}
+
+func NewMetricsCollector(cli client.Client) *MetricsCollector {
+	return &MetricsCollector{
+		client: cli,
+	}
+}
+
+func (mc *MetricsCollector) CollectResourceMetrics(ctx context.Context) error {
```

Comment:
Rather than calling this on every reconcile, we may want to change this into using the `prometheus.Collector` interface, then call `Collect` on each _metrics_ call instead. This ensures up-to-date information for the data we gather.